### ISSUE TITLE
Add support for rqlite

### ISSUE THREAD

**Issue by wchiquito** [neilotoole/sq#444]
**Additional context**
Could use this Go driver: gorqlite (https://github.com/rqlite/gorqlite).

**Comment by neilotoole** [comment]
@wchiquito: I've been tracking `rqlite` for several years, I'd love to add support. (Shout out to my fellow O'Toole in databases, @otoolep... apparently databases run in the family).

The way `sq` is currently built, only DBs that support the Go stdlib `database/sql` driver mechanism can be added. (Although that could be fixed with an appropriate level of abstraction). I see that there is indeed [such a driver for `rqlite`](https://github.com/rqlite/gorqlite?tab=readme-ov-file#driver-for-databasesql), so `rqlite` support seems doable.

I need to write up a "How to write a sq driver" document. At this time, if anybody is interested in starting on `rqlite` support, please reach out.

**Comment by otoolep** [comment]
Thanks @neilotoole -- yes, amusing coincidence. Happy to answer any questions on rqlite if it helps.

Yeah, that driver was added in the last year. As long as you can accept a driver that doesn't support traditional transaction support, it should work I believe (rqlite supports a different type of transaction than what most relational databases support, due to the nature of the HTTP API).

**Comment by otoolep** [comment]
That looks like a dodgy link in https://github.com/neilotoole/sq/issues/444#issuecomment-2442230626

**Comment by neilotoole** [comment]
Quick update: there's now a [`sakiladb/rqlite`](https://hub.docker.com/repository/docker/sakiladb/rqlite) Docker image preloaded with the Sakila dataset, matching the convention of the other `sakiladb/*` images that `sq`'s integration tests already use.

- Image: `docker pull sakiladb/rqlite:latest` (or `:10`; multi-arch, ~70 MB, cosign-signed)
- Source: https://github.com/sakiladb/rqlite
- Default creds: `sakila` / `p_ssW0rd`

Quick smoke test:

```shell
docker run -p 4001:4001 --add-host rqlite1:127.0.0.1 -d sakiladb/rqlite:latest
curl -u sakila:p_ssW0rd 'http://localhost:4001/db/query?level=strong&q=SELECT+count(*)+FROM+actor'
# → {"results":[{"columns":["count(*)"],"types":["integer"],"values":[[200]]}]}
```

A 3-node cluster harness (`cluster-compose.yml`) is also in the repo, so the driver can be exercised against both single-node and clustered rqlite once written.

This unblocks the `drivers/rqlite/` work in `sq` itself — the test-handle pattern from `CONTRIBUTING.md` can now reference `sakiladb/rqlite` directly. The actual Go driver still needs to be written; gorqlite + most of the `drivers/sqlite3/` dialect should be reusable per @otoolep's notes above (with the transaction-semantics caveat).

**Comment by neilotoole** [comment]
@otoolep: this `sq` driver effort for `rqlite` is finally moving forward. There's some dirty decisions to be made, e.g.

> Scheme: `rqlite://` (HTTP) and `rqlites://` (HTTPS), redis-style.

I'll throw out a first-pass version of it in the coming days/weeks. Will mark as beta. Would greatly appreciate your feedback about the impl and ergonomics.

**Comment by otoolep** [comment]
Sounds good, happy to provide input.

**Comment by neilotoole** [comment]
Closed by #735 (squash merge: d13c1ff8 on master, 2026-06-07).

The PR ships the full rqlite driver: read path, inspect metadata, the complete `driver.SQLDriver` write surface (CreateTable, Truncate, CopyTable, NewBatchInsert, PrepareInsertStmt, PrepareUpdateStmt, all four AlterTable* methods), credential redaction across error paths, auto-port-injection (4001), shell completion for `rqlite://` and `rqlites://`, site docs at https://sq.io/docs/drivers/rqlite, and the skill reference for agents.

Eight follow-up issues track deferred work: #737 (faithful CopyTable), #738 (cluster handle), #741 (clickhouse/oracle completion gap), #742 (localhost UX hint), #743 (bare-host completion stage), #744 (Size nullability), plus sakiladb/rqlite#2 (upstream schema divergence) and sakiladb/rqlite#3 (upstream docs).
